Rubberduck.CodeAnalysis/Inspections/Concrete/EnumerationDeclaredWithinWorksheetInspection.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Rubberduck.CodeAnalysis.Inspections.Abstract;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Parsing.VBA.DeclarationCaching;
+using Rubberduck.Resources.Inspections;
+using Rubberduck.VBEditor.ComManagement;
+using Rubberduck.VBEditor.SafeComWrappers;
+
+namespace Rubberduck.CodeAnalysis.Inspections.Concrete
+{
+    internal class PublicEnumerationDeclaredWithinWorksheetInspection : DeclarationInspectionBase
+    {
+        private readonly IProjectsProvider _projectsProvider;
+
+        public PublicEnumerationDeclaredWithinWorksheetInspection(IDeclarationFinderProvider declarationFinderProvider, IProjectsProvider projectsProvider)
+            : base(declarationFinderProvider)
```

Comment:
You should provide the declaration type to the base constructor. This leads to an efficient filtering to only that declaration type.